src/modules/split_text_to_size.js

```diff
@@ -112,18 +112,31 @@
   * @returns {number} result
   */
   var getStringUnitWidth = API.getStringUnitWidth = function (text, options) {
-    options = options || {};
 
-    var fontSize = options.fontSize || this.internal.getFontSize();
-    var font = options.font || this.internal.getFont();
-    var charSpace = options.charSpace || this.internal.getCharSpace();
-    var result = 0;
-    if (typeof font.metadata.widthOfString === "function") {
-      result = font.metadata.widthOfString(text, fontSize, charSpace) / fontSize;
-    } else {
-      result = getArraySum(getCharWidthsArray.apply(this, arguments));
+    self = this;
+    function _getStringUnitWidth_(text, options)
```

Comment:
So you encapsulate but the code is the same?

Reply:
This is true. My investigation shows that JSPdf does not do kerning when placing the text. But considers kerning in getSringUnitWitdth. This is the root cause for the problems with right alignment.